tiledb/sm/filter/xor_filter.cc

```diff
@@ -48,6 +48,35 @@ void XORFilter::dump(FILE* out) const {
   fprintf(out, "XORFilter");
 }
 
+bool XORFilter::accepts_input_datatype(Datatype datatype) const {
+  switch (datatype_size(datatype)) {
+    case sizeof(int8_t):
+    case sizeof(int16_t):
+    case sizeof(int32_t):
+    case sizeof(int64_t):
+      return true;
+    default:
+      return false;
+  }
+}
+
+Datatype XORFilter::output_datatype(tiledb::sm::Datatype input_type) const {
+  switch (datatype_size(input_type)) {
+    case sizeof(int8_t):
+      return Datatype::INT8;
```

Comment:
Yet another example of me not being completely aware of all the details, but is there a concern on the signed-ness of the returned types? Obviously we have two choices of signed or unsigned for each byte width and I don't see an immediate reason we'd go with signed (or unsigned for that matter). I'm mostly just wondering if this would prevent any filters coming after this one because they require unsigned input types?

Reply:
IIUC xor filter will always return signed values based on matching byte width of input type. I based the output datatype here off the `XORFilter::run_forward` function delegate to `XORFilter::xor_part<T>` where T is always a signed integer.

Reply:
I see your point there, though that also just feels like an arbitrary "we picked one" sort of situation. However, reading that code I realize that if we have any issues about types, we can always just add a "Type Conversion Filter" which just exists to allow users to cast datatypes as needed. No need to worry about that now, but now that I can see a solution if we find weirdo edge cases is enough for me to not worry as much about the filter type permutations.